src/actor/npc/ac_npc_shop_master.c

```diff
@@ -0,0 +1,2677 @@
+    if ((int)mDemo_Get_OrderValue(0x4, 0x01) == 0x2) {
+        aNSC_setupAction(shop_master, game, 0x3a);
+    }
+}
+
+static void aNSC_pc_present_trans_takeout(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game) {
+    if (shop_master->npc_class.left_hand.item_actor_p == NULL) {
+        ACTOR *actor = CLIP(handOverItem_clip)->birth_proc(shop_master->password.item, 0x7, 0x1, (ACTOR *)shop_master);
+        if (actor) {
+            CLIP(handOverItem_clip)->chg_request_mode_proc((ACTOR *)shop_master, 0x1);
+            shop_master->npc_class.left_hand.item_actor_p = actor;
+        }
+    }
+    if (shop_master->npc_class.draw.main_animation_state == 0x1) {
+        if (CLIP(handOverItem_clip)->master_actor == (ACTOR *)shop_master) {
+            CLIP(handOverItem_clip)->chg_request_mode_proc((ACTOR *)shop_master, 0x2);
+            aNSC_setupAction(shop_master, game, 0x3b);
+        }
+    }
+}
+
+static void aNSC_pc_present_trans_wait(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game) {
+    if (CLIP(handOverItem_clip)->master_actor != (ACTOR *)shop_master) {
+        aNSC_setupAction(shop_master, game, 0x3c); 
+    }
+}
+
+static void aNSC_pc_present_end_wait(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game) {
+    if (CLIP(handOverItem_clip)->master_actor == NULL) {
+        mMsg_Unset_LockContinue(mMsg_Get_base_window_p());
+        aNSC_setupAction(shop_master, game, 0xf);
+    }
+}
+
+static void aNSC_wait(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game) {
+    if (aNSC_message_ctrl(shop_master, game) != TRUE) {
+        int act = aNSC_decide_next_move_act(shop_master, game);
+        if (act != shop_master->action) {
+            aNSC_setupAction(shop_master, game, act); 
+        } else {
+            aNSM_search_player(shop_master, game);
+        }
+    }
+}
+
+static void aNSC_walk_pl_other_zone(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game) {
+    if (aNSC_message_ctrl(shop_master, game) != TRUE) {
+        int act = aNSC_decide_next_move_act(shop_master, game);
+        if (act != shop_master->action) {
+            aNSC_setupAction(shop_master, game, act);
+        } else {
+            aNSM_search_player2(shop_master, game);
+        }
+    }
+}
+
+static void aNSC_turn(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game) {
+    ACTOR *actorx = (ACTOR *)shop_master;
+    if (aNSC_message_ctrl(shop_master, game) != TRUE) {
+        if (get_player_actor_withoutCheck(game)) {
+            s16 player_angle = shop_master->player_angle;
+            chase_angle(&actorx->shape_info.rotation.y, player_angle, 0x800);
+            actorx->world.angle.y = actorx->shape_info.rotation.y;
+            if (ABS(player_angle - actorx->shape_info.rotation.y) <= 0x4000) {
+                aNSC_setupAction(shop_master, game, 0x3d);
+            }
+        }
+    }
+}
+
+static void aNSC_set_talk_info_goodbye_wait() {
+    mDemo_Set_msg_num(aNSC_get_msg_no(aNSC_MSG_SAY_GOODBYE));
+    mDemo_Set_camera(TRUE);
```

Comment:
This is wrong. mDemo_Set_camera takes the camera process type. So this would be `CAMERA2_PROCESS_NORMAL`.